setup/helpers/puppetserver_helper.rb

```diff
@@ -0,0 +1,644 @@
+    param_from_build(url, "SUITE_COMMIT")
+  end
+
+  def latest_puppet_version
+    url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent%20suite%20pipelines/job/platform_puppet-agent_intn-van-promote_suite-daily-promotion-master/lastSuccessfulBuild/api/json"
+    param_from_build(url, "SUITE_VERSION")
+  end
+
+  def param_from_build(url, param)
+    response = https_request(url, :git)
+    json = JSON.parse(response.body)
+    actions = json["actions"].find { |hash| hash["_class"] == "hudson.model.ParametersAction" }
+    parameters = actions["parameters"]
+    pkg_build_param = parameters.find { |hash| hash["name"] == param }
+    pkg_build_param["value"]
+  end
+
+  # Issue an HTTP request and return the Net::HTTPResponse object. Lifted from
+  # https://github.com/puppetlabs/pe_acceptance_tests/blob/2015.3.x/lib/http_calls.rb
+  # and slightly modified.
+  # url: (String) URL to poke
+  # method: (Symbol) :post, :get
+  # cert: (OpenSSL::X509::Certificate, nil) The certificate to
+  #       use for authentication.
+  # key: (OpenSSL::PKey::RSA, nil) The private key to use for
+  #      authentication
+  # body: (String) Request body (default empty)
+  require 'net/http'
+  require 'uri'
+  def https_request(url, request_method, cert = nil, key = nil, body = nil, options = {})
+    # Make insecure https request
+    uri = URI.parse(url)
+    http = Net::HTTP.new(uri.host, uri.port)
+
+    if cert
+      if cert.is_a?(OpenSSL::X509::Certificate)
+        http.cert = cert
+      else
+        raise TypeError, "cert must be an OpenSSL::X509::Certificate object, not #{cert.class}"
+      end
+    end
+
+    if key
+      if key.is_a?(OpenSSL::PKey::RSA)
+        http.key = key
+      else
+        raise TypeError, "key must be an OpenSSL::PKey:RSA object, not #{key.class}"
+      end
+    end
+
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    http.read_timeout = options[:read_timeout] if options[:read_timeout]
+
+    if request_method == :post
+      request = Net::HTTP::Post.new(uri.request_uri)
+      # Needs the content type even though no data is being sent
+      request.content_type = 'application/json'
+      request.body = body
+    else
+      request = Net::HTTP::Get.new(uri.request_uri)
+    end
+
+    start = Time.now
+    response = http.request(request)
+    stop = Time.now
+
+    logger.debug "Remote took #{stop - start} to respond"
+    response
+  end
+
+  def reload_server(host = master, opts = {})
+    service = options['puppetservice']
+    on(host, "service #{service} reload", opts)
+  end
+
+  def apply_one_hocon_setting(hocon_host,
+                              hocon_file_path,
+                              hocon_setting,
+                              hocon_value)
+    hocon_manifest =<<-EOF.gsub(/^ {6}/, '')
```

Comment:
what is up with the gsub.  Maybe a comment might help.

Reply:
this is very common to remove indentation from heredocs.
https://stackoverflow.com/a/5638187/3010822

i'm not a huge fan of monkey patching, even super-unique method names... 
but the gsub is well established there.  There are similar solutions for bash, etc.